### PR TITLE
cpu/[msp|cc]430: added peripheral timer driver

### PR DESCRIPTION
**Pull request by haukepetersen**
To get RIOT ready for the xtimer, here is the low-level timer implementation for all MSP430 based boards. For backward compatibility, all effected boards do now use the `hwtimer_compat` module.

Please note, that the `vtimer` will not work correctly anymore, as it depends on a 32-bit `hwtimer` (while we now only have 16-bit wide low-level timers). But this should not matter much, as the boards run perfectly with the `xtimer`!

Note2 (especially @kaspar030): So far there is almost identical code for the `cc430` and the `msp430fxyz`. Especially for some other peripherals, they seem to differ quite a bit, but I am not sure yet to which extend. So the strategy is clearly for now: implement separately and merge again as soon as I have a better feeling for those MCUs - so be patient :-)

PRs for other peripheral timers will follow soon...

### CODE REVIEW

**Comment by kaspar030** [comment]
Nice! will test ASAP.

(you probably need to (void) some arguments)

**Comment by haukepetersen** [comment]
@kaspar030, any progress testing this?

**Comment by kaspar030** [comment]
nope, busy, will do later today!

**Comment by kaspar030** [comment]
Works as expected on msb-430h. Can't test on other boards. Do you want to fix the "N" define in xtimer or the msb headers? ACK after that is fixed.

**Comment by haukepetersen** [comment]
Just fixing it as we speak..

**Comment by haukepetersen** [comment]
fixed the xtimer tests that Travis was complaining about and removed the TI legacy header (`legacymsp430.h`). The xtimer test should work fine on all effected boards.

**Comment by kaspar030** [comment]
Tested again on msb-430h, ACK holds from my side if travis is happy.

**Comment by OlegHahm** [comment]
Will test on Z1 and WSN430 now.

**Comment by OlegHahm** [comment]
Hm, on Z1 with `example/timer_periodic_wakeup` I observe a weird pattern (look at the pyterm timestamp):
```
2015-09-02 17:31:06,363 - INFO # slept until 79065481
2015-09-02 17:31:07,348 - INFO # slept until 80065481
2015-09-02 17:31:08,335 - INFO # slept until 81065481
2015-09-02 17:31:08,338 - INFO # slept until 81002729
2015-09-02 17:31:08,341 - INFO # slept until 81005633
2015-09-02 17:31:08,344 - INFO # slept until 81008539
2015-09-02 17:31:08,346 - INFO # slept until 81011447
2015-09-02 17:31:08,349 - INFO # slept until 81014352
2015-09-02 17:31:08,352 - INFO # slept until 81017263
2015-09-02 17:31:08,355 - INFO # slept until 81020174
2015-09-02 17:31:08,358 - INFO # slept until 81023079
2015-09-02 17:31:08,361 - INFO # slept until 81025989
2015-09-02 17:31:08,364 - INFO # slept until 81028893
2015-09-02 17:31:08,367 - INFO # slept until 81031799
2015-09-02 17:31:08,369 - INFO # slept until 81034706
2015-09-02 17:31:08,372 - INFO # slept until 81037615
2015-09-02 17:31:08,375 - INFO # slept until 81040522
2015-09-02 17:31:08,378 - INFO # slept until 81043426
2015-09-02 17:31:08,381 - INFO # slept until 81046332
2015-09-02 17:31:08,384 - INFO # slept until 81049242
...
```
The node seems to run fine for some while and then trigger a lot of interrupts at once, go back to normal mode and so on...

**Comment by haukepetersen** [comment]
I can reproduce this behavior on the `msb-430h`. @kaspar030, any hint if this is `xtimer` related or if is this caused by the peripheral timer?

**Comment by OlegHahm** [comment]
For the record: very similar pattern on wsn430-v1_3b

**Comment by kaspar030** [comment]
same on msb430h. hm, whenever that happens, the actual sleep time goes backwards by around 63k us, which sounds like a 16bit handling issue.

**Comment by kaspar030** [comment]
@haukepetersen yes, let me try to rule out xtimer stuff, but it actually looks like that.

**Comment by haukepetersen** [comment]
yap. Is there maybe some mask define for 16-bit timers that I missed to set for the boards, but xtimer needs?

**Comment by OlegHahm** [comment]
I don't know if related and/or helpful, but `tests/xtimer_usleep_until` terminates at
```
2015-09-02 17:46:34,409 - INFO # Testing interval 95...
2015-09-02 17:46:34,415 - INFO # Testing interval 94...
2015-09-02 17:46:34,416 - INFO # Testing interval 93...
2015-09-02 17:46:34,416 - INFO # Testing interval 92...
2015-09-02 17:46:34,434 - INFO # Testing interval 91...
2015-09-02 17:46:34,435 - INFO # Testing interval 90...
2015-09-02 17:46:34,435 - INFO # Testing interval 89...
2015-09-02 17:46:34,436 - INFO # Testing interval 88...
2015-09-02 17:46:34,436 - INFO # Testing interval 87...
2015-09-02 17:46:34,437 - INFO # Testing interval 86...
2015-09-02 17:46:34,437 - INFO # Testing interval 85...
2015-09-02 17:46:34,437 - INFO # Testing interval 84...
2015-09-02 17:46:34,438 - INFO # Testing interval 83...
2015-09-02 17:46:34,439 - INFO # Testing interval 82...
2015-09-02 17:46:34,439 - INFO # Testing interval 81...
2015-09-02 17:46:34,505 - INFO # Testing interval 80...
2015-09-02 17:46:34,572 - INFO # Testing interval 79...
2015-09-02 17:46:34,639 - INFO # Testing interval 78...
2015-09-02 17:46:34,705 - INFO # Testing interval 77...
2015-09-02 17:46:34,772 - INFO # Testing interval 76...
2015-09-02 17:46:34,839 - INFO # Testing interval 75...
2015-09-02 17:46:34,905 - INFO # Testing interval 74...
2015-09-02 17:46:34,972 - INFO # Testing interval 73...
2015-09-02 17:46:35,039 - INFO # Testing interval 72...
2015-09-02 17:46:35,105 - INFO # Testing interval 71...
```

File: boards/chronos/include/periph_conf.h [hunk 1]
```diff
@@ -14,17 +15,28 @@
  * @brief       Chronos peripheral configuration
  *
  * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
  */
 
-#ifndef PERIPH_CONF_H_
-#define PERIPH_CONF_H_
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
```
Comment (OlegHahm) [inline boards/chronos/include/periph_conf.h]: minor: but why did you revert this change?

Comment (haukepetersen) [inline boards/chronos/include/periph_conf.h]: propably copy+paste. I tend to use guards without trailing underscores in the files I create...

Comment (OlegHahm) [inline boards/chronos/include/periph_conf.h]: No strong feelings, would just vote for consistency and currently we have 36 `PERIPH_CONF_H_`s and only 6 `PERIPH_CONF_H`s.

Comment (haukepetersen) [inline boards/chronos/include/periph_conf.h]: I don't care, I can offer you to open a follow-up PR to make them all consistent. Would that work?

Comment (OlegHahm) [inline boards/chronos/include/periph_conf.h]: works for me

**Comment by haukepetersen** [comment]
re-introduced `eINT` and `dINT`, as it seems that the `pthread` implementations still use them and Travis fails without them...

**Comment by kaspar030** [comment]
> re-introduced eINT and dINT, as it seems that the pthread implementations still use them and Travis fails without them...

Uff, that should seriously be fixed in pthreads. Promise to at least create an issue. ;)

**Comment by kaspar030** [comment]
I have the feeling that this somehow loses interrupts, but I can't pinpoint it. I'm on it.

**Comment by kaspar030** [comment]
No, my problem is xtimer related. ACK when travis is happy!

**Comment by haukepetersen** [comment]
nice :-)

**Comment by haukepetersen** [comment]
Travis is happy -> and go!

**Comment by kaspar030** [comment]
yay!
